autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/settings/ImportGooglePasswordMainAppSettings.kt

```diff
@@ -0,0 +1,140 @@
+import com.duckduckgo.autofill.impl.store.InternalAutofillStore
+import com.duckduckgo.autofill.impl.ui.credential.management.importpassword.google.ImportFromGooglePasswordsDialog
+import com.duckduckgo.autofill.impl.ui.credential.management.importpassword.google.ImportFromGooglePasswordsDialog.ImportPasswordsDialog
+import com.duckduckgo.autofill.impl.ui.credential.management.importpassword.google.ImportFromGooglePasswordsDialog.ImportPasswordsDialog.Companion.KEY_IMPORT_SUCCESS
+import com.duckduckgo.common.ui.menu.PopupMenu
+import com.duckduckgo.common.ui.view.gone
+import com.duckduckgo.common.ui.view.listitem.OneLineListItem
+import com.duckduckgo.common.ui.view.show
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.settings.api.CompleteSetupSettingsPlugin
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import logcat.logcat
+
+@ContributesMultibinding(ActivityScope::class)
+@PriorityKey(100)
+class ImportGooglePasswordMainAppSettings @Inject constructor(
+    private val pixel: Pixel,
+    private val autofillStore: InternalAutofillStore,
+    private val dispatchers: DispatcherProvider,
+    private val promoRules: InSettingsPasswordImportPromoRules,
+    private val activity: AppCompatActivity,
+) : CompleteSetupSettingsPlugin {
+
+    override fun getView(context: Context): View {
+        return OneLineListItem(activity).apply {
+            setLeadingIconResource(R.drawable.key_color_import_24)
+
+            setPrimaryText(activity.getString(R.string.importPasswordsChooseMethodDialogTitle))
+            setOnClickListener {
+                onLaunchImportFlow(activity, this)
+            }
+
+            configureOverflowMenu()
+
+            gone()
+        }.also {
+            it.showIfPermitted()
+        }
+    }
+
+    private fun OneLineListItem.showIfPermitted() {
+        activity.lifecycleScope.launch {
+            if (promoRules.canShowPromo()) {
+                show()
+            }
+        }
+    }
+
+    private fun onLaunchImportFlow(activity: AppCompatActivity, rootView: View) {
+        listenForImportResult(activity, rootView)
+
+        val dialog = ImportFromGooglePasswordsDialog.instance(importSource = MainAppSettings, tabId = DIALOG_TAG)
+        dialog.show(activity.supportFragmentManager, DIALOG_TAG)
+
+        val params = mapOf("source" to MainAppSettings.value)
+        pixel.fire(AUTOFILL_IMPORT_GOOGLE_PASSWORDS_EMPTY_STATE_CTA_BUTTON_TAPPED, params)
+    }
+
+    private fun listenForImportResult(activity: AppCompatActivity, rootView: View) {
+        val resultKey = ImportPasswordsDialog.resultKey(tabId = DIALOG_TAG)
+
+        activity.supportFragmentManager.setFragmentResultListener(resultKey, activity) { _: String, result: Bundle ->
```

Comment:
Note: I feel all these fragmentResultListeners are a bit weak in terms that relies on how we launch and close the import flows, and they can be tricky to debug if they don't behave as expected. We could expand `AutofillEffectDispatcher` to emit import complete events and observe them from other parts, which is under our control.

Reply:
Treating this as non-blocking and something we can discuss offline.